lib/home/profile/account/change_password.dart

```diff
@@ -1,17 +1,199 @@
+    if (_isTestingMode) {
+      setState(() => _isCurrentPasswordCorrect = true);
+      return;
+    }
+    // to here
+
+    final user = FirebaseAuth.instance.currentUser;
+    final cred = EmailAuthProvider.credential(
+      email: user!.email!,
+      password: _currentPasswordController.text,
+    );
+
+/*
+    try {
+      await user.reauthenticateWithCredential(cred);
+      setState(() => _isCurrentPasswordCorrect = true);
+    } catch (e) {
+      setState(() => _isCurrentPasswordCorrect = false);
+    }
+  }
+*/
+
+    // Delete from here
+    try {
+      await user.reauthenticateWithCredential(cred);
+      if (mounted) {
+        setState(() => _isCurrentPasswordCorrect = true);
+      }
+    } catch (e) {
+      if (mounted) {
+        setState(() => _isCurrentPasswordCorrect = false);
+      }
+    }
+  }
+  // to here
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
         title: const Text('Change Password'),
+        // Delete from here
+        actions: <Widget>[
+          IconButton(
+            icon: Icon(_isTestingMode ? Icons.toggle_on : Icons.toggle_off),
+            onPressed: () => setState(() => _isTestingMode = !_isTestingMode),
+          ),
+        ],
+        // to here
       ),
-      body: const Center(
-        child: Text('Change password page put here'),
+      body: Center(
+        child: Container(
+          padding: const EdgeInsets.all(16.0),
+          width: 500,
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              _buildPasswordField(
+                controller: _currentPasswordController,
+                labelText: 'Current Password',
+                isCorrect: _isCurrentPasswordCorrect,
+              ),
+              const SizedBox(height: 30),
+              _buildPasswordField(
+                controller: _newPasswordController,
+                labelText: 'New Password',
+                enabled: _isCurrentPasswordCorrect,
+              ),
+              _buildPasswordField(
+                controller: _confirmPasswordController,
+                labelText: 'Confirm New Password',
+                enabled: _isCurrentPasswordCorrect,
+              ),
+              ElevatedButton(
+                onPressed: _isCurrentPasswordCorrect ? _changePassword : null,
+                child: const Text('Change Password'),
+              ),
+            ],
+          ),
+        ),
       ),
     );
   }
+
+  Widget _buildPasswordField({
+    required TextEditingController controller,
+    required String labelText,
+    bool isCorrect = false,
+    bool enabled = true,
+  }) {
+    return TextField(
+      controller: controller,
+      obscureText: true,
+      enabled: enabled,
+      decoration: InputDecoration(
+        labelText: labelText,
+        labelStyle: const TextStyle(color: Colors.black),
```

Comment:
Instead of using `_validateCurrentPassword()` to determine whether the "change password" button is enabled, we could call it when the button is tapped, and then use this `InputDecoration` to set an error message:

```suggestion
      decoration: InputDecoration(
        labelText: labelText,
        labelStyle: const TextStyle(color: Colors.black),
        errorText: isCorrect ? null : 'please double-check the password and try again',
```

If something like this is implemented, it might make sense for `isCorrect` to default to `true`, and then be set to `false` when there's an error.